### PR TITLE
[RfC]: Add Support for Multiple Snyk Organizations

### PR DESCRIPTION
**Pull request by Silthus**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk-tech-services/general/wiki/Tests)
- [ ] Commit history is tidy & follows Contributing guidelines [ℹ︎](./CONTRIBUTING.md#commit-messages)

### What this does

One big feature that is currently missing from the plugin is the support for multiple Snyk organizations. This PR adds that feature in a backwards compatible and minimal way.

It adds a new `snyk.io/org-ids` annotations that allows defining a comma separated list of organization ids. It will then iterate over those organizations and tries to find any of the specified targets or projects within any of those organizations, aggregating them into a single list.

In addition the link to the individual tabs changes when multiple orgs are specified to include the org slug in the url. This avoids duplicate links.

### Notes for the reviewer

I also upgraded all dependencies to the latest backstage version, as I was unable to build the project otherwise. The commits are cleanly separated.

Don't hesitate to reach out if you have any additional requests or things your would like to change.

### More information

This PR is supported by the Interhyp as part of the ING.

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const snykApiRef: ApiRef<SnykApi> = createApiRef<SnykApi>({
 export interface SnykApi {
   ListAllAggregatedIssues(orgName: string, projectId: string): Promise<any>;
   ProjectDetails(orgName: string, projectId: string): Promise<any>;
-  GetCompleteProjectsListFromAnnotations(orgId: string, annotations: Record<string,string>):Promise<ProjectsData[]>;
+  GetCompleteProjectsListFromAnnotations(orgId: string, annotations: Record<string,string>, ignoreMissingTargets: boolean):Promise<ProjectsData[]>;
```
Comment (aarlaud) [inline src/api/index.ts:51]: What is the ignoreMissingTargets doing/for? I'm not sure I understand the situation where that'd be needed.

**Comment by aarlaud** [comment]
For clarity since there might be multiple orgs, I think it'd be useful to add in the tab metadata the org slug+ID. Simply adding in SnykTab.tsx the following 

```
const metadata = {
      origin: value.projectDetails.origin,
      type: value.projectDetails.type,
      "created": value.projectDetails.created,
      "last tested": value.projectDetails.lastTestedDate,
      "Project ID": `${value.projectDetails.id}`,
      "Organization": `${orgId}`
    };
```

image

What do you think?

File: src/api/index.ts (hunk 1 above)
Comment (Silthus) [reply src/api/index.ts:51]: The target resolution throws an error if a target slug/id is not found within an organization. However with multiple orgs there will always be a case where a target is not found within an org.
The relevant line is this one here: https://github.com/snyk-tech-services/backstage-plugin-snyk/pull/167/files#diff-764761d79ed791ea154187526b088322dba3bebe94f9d38ab22120554aba77c7R179

And this is where the exception is thrown if a target is not found within an org: https://github.com/Interhyp/backstage-plugin-snyk/blob/1ad9438bec685daa30d9b2d61741f9bea14f4223/src/api/index.ts#L253

**Comment by Silthus** [comment]
> Alright @Silthus, can you rebase off develop ? thanks

Ok I rebased. 
Maybe it would be a good idea to add prettier and eslint to the project to get rid of the code style diffs and make formatting easier?

**Pull request merged**